android-components/components/feature/search/src/test/java/mozilla/components/feature/search/middleware/SearchMiddlewareTest.kt

```diff
@@ -1243,6 +1243,42 @@ class SearchMiddlewareTest {
         assertFalse(store.state.search.disabledSearchEngineIds.contains("bing"))
     }
 
+    @Test
+    fun `WHEN restore hidden search engines action THEN hidden engines are added back to bundled engines list`() {
```

Comment:
This test (or a separate new one) should assert that what we put into the BrowserStore via the `SearchReducer` matches with what we end passing to the `MetadataStorage`.

These are two systems that need to always be in sync, so we should ensure we do not regress that behaviour when we are going through two separate code paths for transforming our search engine lists.